src/base/random.h

```diff
@@ -37,6 +41,12 @@ struct RNG {
 	void read_state(StreamRead&);
 	void write_state(StreamWrite&);
 
+	static uint32_t static_rand();
+	static inline uint32_t static_rand(const uint32_t exclusive_upper_bound) {
+		assert(exclusive_upper_bound > 1);
```

Comment:
This cannot be guaranteed. I just got an assertion error while loading a theme with just one background (called in main.cc l.256. Also x % 1 should not be a problem.